### DISCUSSION TITLE
How to get the response body inside a wrap_fn middleware

### DISCUSSION THREAD

**Discussion by Franzx41** [actix/actix-web#3359] [Q&A] [answered]
Hi, guys. How I can get the response body string from the middleware inside wrap_fn?

```
use actix_web::{ web, App, HttpServer, Responder, dev::Service as _ };

async fn index() -> impl Responder {
    "API is up"
}

#[actix_web::main]
async fn main() -> std::io::Result<()> {

    HttpServer::new(move || {
        App::new()
            .wrap_fn(|req, srv| {
                let fut = srv.call(req);
                async {
                    let res = fut.await?;
                    // How to get the response body here?
                    Ok(res)
                }
            })
            .route("/", web::get().to(index))
    })
    .bind("0.0.0.0:7878")?
    .run()
    .await
}
```

**Comment by robjtede** [comment, marked as answer]
The highlighted lines from this example show how to read the response body: https://github.com/actix/examples/blob/master/middleware/encrypted-payloads/src/main.rs#L127-L136

Though it's important to put a body back into the response so it actually gets sent out as well: https://github.com/actix/examples/blob/master/middleware/encrypted-payloads/src/main.rs#L153-L155

I'd also recommend using [`from_fn()` from `actix-web-lab`](https://docs.rs/actix-web-lab/latest/actix_web_lab/middleware/fn.from_fn.html) which is a much better developer experience than wrap_fn.
